File: web/src/utils/seasonLabels.ts
const compactPattern = /^(\d{4})\/(\d{2})$/
const longPattern = /^(\d{4})\/(\d{4})$/

export function seasonLabelFromStartYear(startYear: number): string {
  if (!Number.isInteger(startYear) || startYear < 0) {
    throw new Error('startYear must be a non-negative integer')
  }
  return `${startYear}/${String((startYear + 1) % 100).padStart(2, '0')}`
}

export function toCompactSeasonLabel(label: string): string {
  const raw = label.trim()

  const compactMatch = compactPattern.exec(raw)
  if (compactMatch) {
    const startYear = Number(compactMatch[1])
    const endTwoDigits = Number(compactMatch[2])
    if (endTwoDigits !== (startYear + 1) % 100) {
      throw new Error('Invalid season label rollover')
    }
    return seasonLabelFromStartYear(startYear)
  }

  const longMatch = longPattern.exec(raw)
  if (longMatch) {
    const startYear = Number(longMatch[1])
    const endYear = Number(longMatch[2])
    if (endYear !== startYear + 1) {
      throw new Error('Invalid season label rollover')
    }
    return seasonLabelFromStartYear(startYear)
  }

  throw new Error('Season label must use YYYY/YY or YYYY/YYYY format')
}

export function toLongSeasonLabel(label: string): string {
  const compact = toCompactSeasonLabel(label)
  const startYear = Number(compact.split('/')[0])
  return `${startYear}/${startYear + 1}`
}

File: src/beta_engine/domain/calendar/season_labels.py
"""Season label compatibility utilities.

Canonical internal label for the new Season Registry is compact ``YYYY/YY``.
This module provides additive compatibility helpers so legacy long labels
(``YYYY/YYYY``) continue to work without broad runtime behavior changes.
"""

from __future__ import annotations

import re

_COMPACT_RE = re.compile(r"^(\d{4})/(\d{2})$")
_LONG_RE = re.compile(r"^(\d{4})/(\d{4})$")


def season_label_from_start_year(start_year: int) -> str:
    if not isinstance(start_year, int):
        raise ValueError("start_year must be an integer")
    if start_year < 0:
        raise ValueError("start_year must be non-negative")
    return f"{start_year}/{(start_year + 1) % 100:02d}"


def long_season_label_from_start_year(start_year: int) -> str:
    if not isinstance(start_year, int):
        raise ValueError("start_year must be an integer")
    if start_year < 0:
        raise ValueError("start_year must be non-negative")
    return f"{start_year}/{start_year + 1}"


def season_start_year_from_label(label: str) -> int:
    normalized = normalize_season_label(label)
    match = _COMPACT_RE.match(normalized)
    assert match is not None
    return int(match.group(1))


def normalize_season_label(label: str) -> str:
    if not isinstance(label, str):
        raise ValueError("season label must be a string")
    raw = label.strip()

    compact = _COMPACT_RE.match(raw)
    if compact:
        start_year = int(compact.group(1))
        end_two_digits = int(compact.group(2))
        expected = (start_year + 1) % 100
        if end_two_digits != expected:
            raise ValueError("season label rollover is invalid")
        return season_label_from_start_year(start_year)

    long_label = _LONG_RE.match(raw)
    if long_label:
        start_year = int(long_label.group(1))
        end_year = int(long_label.group(2))
        if end_year != start_year + 1:
            raise ValueError("season label rollover is invalid")
        return season_label_from_start_year(start_year)

    raise ValueError("season label must use YYYY/YY or YYYY/YYYY format")


def to_compact_season_label(label: str) -> str:
    return normalize_season_label(label)


def to_long_season_label(label: str) -> str:
    return long_season_label_from_start_year(season_start_year_from_label(label))
